src/test/java/teammates/test/pageobjects/StudentProfilePage.java

```diff
@@ -158,6 +159,10 @@ private void ensureGenderIsSelectedAs(String gender) {
     public void uploadPicture() {
         uploadPictureSubmit.click();
         browser.selenium.waitForPageToLoad("10000");
+        
+        // Intended compounded explicit wait due to .click()'s inconsistency in triggering selenium's
+        // waitForPageToLoad. Do not remove, can cause a higher rate of random failure in this test.
+        ThreadHelper.waitFor(1000);
```

Comment:
Do we still need this? I thought if we retry the status check, there's no need to wait here.